### PR TITLE
Added currentIndex, selectedIndex and indicatorMaxCount properties for Carousel Widget

### PR DESCRIPTION
**Pull request by vinothvino42**
In this PR, I've fixed a few issues and updated the schema

**1. Index Updates:** Ticket: #949 and #955

1. `currentIndex`
    - For setting the initial index of the carousel which starts the carousel from that child of index.
    - To get the current index which is displayed in the carousel
    - Default (0)
2. `selectedIndex` 
    - To get the selectedIndex only when the user clicks on the carousel item otherwise it'll return -1. Default (-1)
3. `indicatorMaxCount` 
    - This property shows a limited number (for ex: 4) of **carousel indicator** when creating the carousel with more (for ex: 100+) items. 
    - Default (null), if this property is null the indicator works normally how it works before (based on the currentIndex property)

**2. Indicator Alignment and Padding** - Ticket: [#952](https://github.com/EnsembleUI/ensemble/issues/952)
- Here, I added multiple alignment (topLeft, bottomCenter, etc) support for the carousel indicator and replaced the `indicatorOffset` with the `indicatorPadding` property

### CODE REVIEW

**Review by vusters — CHANGES_REQUESTED** [review]
Careful with this. selectedItemIndex is used to know if you click on a card. By setting it to 0 you cannot click on the first item. Please re-do this.

**Comment by vinothvino42** [comment]
> Careful with this. selectedItemIndex is used to know if you click on a card. By setting it to 0 you cannot click on the first item. Please re-do this.

We're using this selectedItemIndex in different scenarios.
1. For setting the initial index of the carousel which starts the carousel from that child of index.
2. To get the selectedItemIndex when the user clicks on the card
3. To get the current index which is displaying in the carousel

Looks like it's confusing. Can we expose two properties?

currentIndex - Which will be used for the first and the second point above. 0 is the default value if it's not given by the user
selectedItemIndex - It'll return the value only if the card selected or else it'll have -1 as the value

**Comment by vusters** [comment]
> > Careful with this. selectedItemIndex is used to know if you click on a card. By setting it to 0 you cannot click on the first item. Please re-do this.
> 
> @vusters We're using this selectedItemIndex in different scenarios.
> 
> 1. For setting the initial index of the carousel which starts the carousel from that child of index.
> 2. To get the current index which is displaying in the carousel
> 3. To get the selectedItemIndex only when the user clicks on the card
> 
> Looks like it's confusing. Can we expose two properties?
> 
> `currentIndex` - Which will be used for the first and the second point above. 0 is the default value if it's not given by the user `selectedItemIndex` - It'll return the value only if the card selected or else it'll have -1 as the value

Ok, good ideal. Keep currentIndex internal for us to track, and a `selectedIndex` to expose which card is selected.

**Comment by vinothvino42** [comment]
@vusters Can you review this PR again? See the description for the new changes

**Comment by kmahmood74** [comment]
@vusters need your review on this one as well as Vinoth made changes

File: assets/schema/ensemble_schema.json [hunk 1]
```diff
@@ -2812,8 +2812,15 @@
                       "type": "string",
                       "description": "Where to display the indicator if specified",
                       "enum": [
```
Comment (vusters) [inline assets/schema/ensemble_schema.json:2814]: don't hardcode this. There must be an alignment schema that we declare already. If not then create it.

File: lib/widget/carousel.dart [hunk 2]
```diff
@@ -464,4 +463,37 @@ enum IndicatorType {
   custom,
 }
 
-enum IndicatorPosition { bottom, top }
+enum IndicatorPosition {
```
Comment (vusters) [inline lib/widget/carousel.dart:466]: why create another enum? Just use Alignment?

File: lib/widget/carousel.dart [hunk 3]
```diff
@@ -149,7 +148,9 @@ class MyController extends BoxController {
   // for multi view this dispatch when clicking on a card
   EnsembleAction? onItemTap;
   EnsembleAction? onItemChange;
-  int selectedItemIndex = -1;
```
Comment (vusters) [inline lib/widget/carousel.dart:152]: So whoever used selectedItemIndex previously will have their page ceased to work? Please make sure our App Developers are aware of this so they can make changes.

File: assets/schema/ensemble_schema.json [hunk 4]
```diff
@@ -2819,9 +2819,17 @@
                       "type": "boolean",
                       "description": "Determines if carousel should loop infinitely or be limited to item length. Default (false)"
                     },
-                    "selectedItemIndex": {
+                    "selectedIndex": {
```
Comment (vusters) [inline assets/schema/ensemble_schema.json:2840]: I thought we agreed one will be internal and the other to be public. I'm looking at this as a customer and I don't understand the difference between selectedIndex and currentIndex.

Comment (vinothvino42) [reply assets/schema/ensemble_schema.json:2840]: `currentIndex` - Which will be used for the first and the second point above. 0 is the default value if it's not given by the user
`selectedIndex` - It'll return the value only if the card selected or else it'll have -1 as the value

File: lib/widget/carousel.dart (hunk 3 above)
Comment (vinothvino42) [reply lib/widget/carousel.dart:152]: Yes, they need to change it to `selectedIndex`

File: lib/widget/carousel.dart (hunk 2 above)
Comment (vinothvino42) [reply lib/widget/carousel.dart]: Done

File: assets/schema/ensemble_schema.json (hunk 1 above)
Comment (vinothvino42) [reply assets/schema/ensemble_schema.json]: Done

**Comment by vinothvino42** [comment]
@vusters I updated the carousel indicatorPosition with using `getAlignment` from the Utils and removed IndicatorPosition enum. Also updated the indicatorPosition of carousel schema with `type-alignment` payload.

Can you review it again?

**Review by vusters — APPROVED** [review]

**Pull request merged**